### DISCUSSION TITLE
fallback value  in `-l` flag if field does not exist

### DISCUSSION THREAD

**Discussion by hansbogert** [cue-lang/cue#3245] [Q&A] [answered]
When dealing with kubernetes manifests it is quite handy to import them with something like:

```
cue import yaml -l 'metadata.namespace' ...
```
However, this fails with Cluster wide resources as there is no namespace defined.
Is there a way to fallback to predefined a value? so conceptually as the following:

```
cue import yaml -l '*metadata.namespace| "_cluster"' ...
```

**Comment by hansbogert** [comment, marked as answer]
It starts making sense to translate `cue export` into `cue transform` in my head, we can just pre-process:

```
cue export yaml ... --out yaml schema.cue foo.yaml | cue import yaml: -l 'namespace' -
```

where schema.cue has:

```cuelang
metadata.namespace: *"_cluster"| string
```

It's a bit unwieldy but works.

**Reply by myitcv** [reply]
Another way to "hold" this would be to use the new `@embed`: https://github.com/cue-lang/cue/discussions/3264

**Reply by hansbogert** [reply]
~Not entirely sure how that would look, but will look into it~ this looks promising. Will follow up with a new answer
